Assets/Scripts/Tags/Hazard.cs

```diff
@@ -1,45 +1,41 @@
 using UnityEngine;
+using UnityEngine.Serialization;
 
 public class Hazard : MonoBehaviour {
+    [Tooltip("Defines how high and low this object will bob up and down, measured in world-space units.")]
+    [SerializeField] private float amplitude = 0.2f;
 
-    [SerializeField] private float verticalSpeed;
-    [SerializeField] private float amplitude;
+    [Tooltip("The rate at which this object will rotate over time, measured in degrees per second.")]
+    [FormerlySerializedAs("rotSpeed")]
+    [SerializeField] private float angularSpeed = 60;
 
-    [SerializeField] private float rotSpeed;
+    [Tooltip("Defines the range within which a vertical speed will be randomly chosen for this object.\n\n" +
+        "This object's vertical speed describes the frequency of the up and down bobbing motion.")]
+    [SerializeField] private Vector2 verticalSpeedRange = new Vector2(2, 3);
 
-    private Vector3 origPos;
-    private Vector3 tempPos;
-
-    private Vector3 rot = new Vector3(45f, 45f, 45f);
-
-    private float rotDegreesPerSecond = 45f;
-
-    private static System.Random rndFloat = new System.Random();
-    private static System.Random rndRot = new System.Random();
-
-    private void Start () {
-        tempPos = origPos = new Vector3(transform.position.x, (float) rndFloat.NextDouble() + .3f, transform.position.z);
-        verticalSpeed = (float)rndFloat.NextDouble();
-
-        //rot = Quaternion.Euler((float)rndRot.NextDouble(), (float)rndRot.NextDouble(), (float)rndRot.NextDouble());
+    private Vector3 originalPosition;
+    private float verticalSpeed;
+    private Vector3 rotationalAxis;
 
+    private void Start() {
+        originalPosition = transform.position;
+        verticalSpeed = Random.Range(verticalSpeedRange.x, verticalSpeedRange.y);
+        rotationalAxis = Random.onUnitSphere;
```

Comment:
This is just super convenient.

We could think through how to do the intricacies of random rotation ourselves, but I just remembered I could use Unity to generate a random axis with better equal probability than me just crudely generating a new Vector3 (random X, Y, Z values) and normalizing it.

I then just rotate around this axis for the entirety of gameplay, since this is only calculated once on start.

Reply:
Oh wow! I did see someone using that Random method before. Although it seems like `onUnitSphere` is a property in Unity's Random class?

Reply:
Exactly!

Reply:
Just like the static vs. [non-static discussion](https://linear.app/2kpixelstudios/issue/DJQ-45#comment-4e72b6a2) (on DJQ-45),

You'll notice something similar with methods vs. properties.
You can achieve the same effect with both in SOME cases.

Because a getter property is like a parameterless getter method.
A setter property is like a 1-parameter setter method.

This is why properties were created -- to simplify the syntax for these basic getter/setter use cases of writing methods, and act as kind of a glue between methods (ACTIONS) and fields (DATA). So in Java, you would HAVE to write getter/setter methods, because Java doesn't have properties in its language syntax. However, C# (since like C# 4 or maybe even earlier), they've allowed us to write a similar, but shorter syntax, called properties, that simplify the writing of getter/setter methods so we don't have to do it that way anymore

```java
//Java
public class Enemy {
    private String name;

    public Enemy(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public void setName(String value) {
        name = value;
    }
}
```


```cs
//C#
public class Enemy {
    private string name;

    //NOTE: I typically put my C# properties right after my fields, because they're (functionally) in between fields and methods:
    public string Name {
        get { return name; }
        set {
            //NOTE: value is a keyword, and is only usable inside the "set" block of a property.
            name = value;

            //NOTE: Here, you can update any other data that must update when this field is set.
        }
    }

    public Enemy(string name) {
        this.name = name;
    }

    //THIS IS POSSIBLE, but I would 100% prefer using C# properties instead, since they are cleaner syntax, and describe the intentions better. Properties are meant for read AND/OR write access to fields, AND quick, convenient calculations that don't even need to be stored as fields.
    public string GetName() {
        return name;
    }

    public void SetName(string value) {
        name = value;
    }
}
```